tests/core/util/test_file_keyring_synchronization.py

```diff
@@ -108,27 +108,26 @@ def child_writer_dispatch_with_readiness_check(
 
     assert remaining_attempts >= 0
 
-    try:
-        with acquire_writer_lock(lock_path, timeout, max_iters):
-            result = func()
-            return result
-    except FileKeyringLockTimeout as e:
-        log.warning(
-            f"[pid:{os.getpid()}] caught exception in child_writer_dispatch_with_readiness_check: "
-            f"FileKeyringLockTimeout {e}"
-        )
-        raise e
-    except Exception as e:
-        log.warning(
-            f"[pid:{os.getpid()}] caught exception in child_writer_dispatch_with_readiness_check: "
-            f"type: {type(e)}, {e}"
-        )
-        raise e
-    finally:
-        # Write out a file indicating this process has completed its work
-        finished_file_path: Path = finished_dir / f"{os.getpid()}.finished"
-        with open(finished_file_path, "w") as f:
-            f.write(f"{os.getpid()}\n")
+    while attempts > 0:
+        log.warning(f"{path}, attempt {attempts}")
+        attempts -= 1
+        try:
+            with Lockfile.create(path, timeout):
+                result = func()
+                return result
+        except Exception as e:
+            log.warning(
+                f"[pid:{os.getpid()}] caught exception in child_writer_dispatch_with_readiness_check: "
+                f"type: {type(e)}, {e}"
+            )
+            continue
+        finally:
+            if attempts == 0:
+                raise LockfileError()
+            # Write out a file indicating this process has completed its work
+            finished_file_path: Path = finished_dir / f"{os.getpid()}.finished"
+            with open(finished_file_path, "w") as f:
+                f.write(f"{os.getpid()}\n")
```

Comment:
this looks a bit odd. You do this every time we fail to lock the file.
The `finally` block is executed right after the `continue` on line 123, right?